Makefile.in

```diff
@@ -14,7 +14,9 @@ CPU = @target_cpu@
 TARGET = @target_alias@
 SYMBOL_PREFIX = @SYMBOL_PREFIX@
 CFLAGS = @CFLAGS@
-INCLUDES = -I$(srcdir) @bfd_include_dir@ @binutils_include_dir@ @zlib_include_dir@
+INCLUDES = -I$(srcdir) -Ibfd-headers @zlib_include_dir@
+BFD_INCLUDE_DIR = $(shell cd "@bfd_include_dir@" && pwd)
```

Comment:
Ok, changed to `:=` assignments which will happen only once.

As to portability, `$(shell ..)` is no less portable than `$(strip ...)`, `$(findstring ...)` or `ifeq`. Neither of those are in POSIX either, yet you didn't hesitate to use them.